docs/inputs.md

```diff
@@ -23,16 +23,38 @@ note: it does not implement [carbon2.0](http://metrics20.org/implementations/)
 
 ## Kafka-mdm (recommended)
 
-The Kafka input supports 2 formats:
-* MetricData Messagepack-encoded (legacy: slow and verbose. they contain the data points as well as all metric data. see #876)
-* MetricPoint messages. (more optimized: contains only id, value and timestamp. see #876)
-
-See the [schema repository](https://github.com/raintank/schema) for more details.
-
 This is the recommended input option if you want a queue. It also simplifies the operational model: since you can make nodes replay data
 you don't have to reassign primary/secondary roles at runtime, you can just restart write nodes and have them replay data, for example.
 Note that [carbon-relay-ng](https://github.com/graphite-ng/carbon-relay-ng) can be used to pipe a carbon stream into Kafka.
 
+The Kafka input supports 2 formats:
+
+* MetricData
+* MetricPoint
+
+Both formats have a corresponding implementation in the [schema repository](https://github.com/raintank/schema), making it trivial
+to implement your own producers (and consumers) if you use Golang.
+
+### MetricData
+
+This format contains the data points as well as all metric identity data and metadata, in messagepack encoded messages (not JSON).
+This format is rather verbose and inefficient to encode/decode.
+See the [MetricData](https://godoc.org/github.com/raintank/schema#MetricData) documentation for format and implementation details.
+
+### MetricPoint
+
+This format is a hand-written binary format that is much more compact and fast to encode/decode compared to MetricData.
+See the [MetricPoint](https://godoc.org/github.com/raintank/schema#MetricPoint) documentation for format and implementation details.
+It is a minimal format that only contains the series identifier, value and timestamp.
+As such, it is paramount that MetricPoint messages for each series has been preceded by a MetricData message for that series.
```

Comment:
s/has been preceded/have been preceded/

Reply:
The true requirement is just that an entry exists in the index for the id of the metricPoint.